pages/desktop/developers/manage_versions.py

```diff
@@ -20,6 +23,27 @@ def wait_for_page_to_load(self):
     def version_approval_status(self):
         return self.find_element(*self._version_approval_status_locator)
 
+    def wait_for_version_autoapproval(self, value):
+        """Method that verifies if auto-approval occurs within a set time"""
+        timeout_start = time.time()
+        # auto-approvals should normally take ~5 minutes;
+        # set a loop to verify if approval occurs within this interval
+        while time.time() < timeout_start + 300:
+            # refresh the page to check if the status has changed
+            self.selenium.refresh()
+            if value not in self.version_approval_status.text:
+                # wait 30 seconds before we refresh again
+                time.sleep(30)
+            # break the loop of the status changed to the expected value within set time
+            else:
+                break
+        # if auto-approval took longer than normal, we want to fail the test and capture the final status
+        if time.time() > timeout_start + 300:
+            pytest.fail(
+                f'Autoapproval took longer than normal; '
+                f'Addon final status was "{self.version_approval_status.text}" instead of "{value}"'
+            )
```

Comment:
You can just use ```else``` here instead of another if statement. https://www.pythontutorial.net/python-basics/python-while-else/

Reply:
Good tip! I've made he update.